content/concepts/multiplex/overview.md

```diff
@@ -9,24 +9,45 @@ aliases:
 
 ## Overview
 
-Stream Multiplexing (_stream muxing_) is a way of sending multiple streams of data over one
-communication link. It combines multiple signals into one unified signal so it can be transported
-'over the wires', then it is demulitiplexed (_demuxed_) so it can be output and used by separate
-applications. This is done to share a single TCP connection using unique port numbers to distinguish
-streams, between the multiple proceeses (such as kademlia and gossipsub) used by applications (such as IPFS)
-to make connection and transmission more efficient.
-
-With muxing, libp2p applications may have many separate streams of communication between peers, as well as
-have multiple concurrent streams open at the same time with a peer. Stream multiplexing allows us to initialize
-and use the same [transport](/concepts/transports/overview) connection across the lifetime of our interaction with a
-peer. With muxing, we also only need to deal with [NAT traversal](/concepts/nat/overview) once to be able to open as
-many streams as we need, since they will all share the same underlying transport connection. Applications can enable
-support for multiple multiplexers, which will allow you to fall back to a widely-supported multiplexer if a preferred
-choice is not supported by a remote peer.
+Network protocols often use bi-directional message streams built on top of a stream
+abstraction to transmit data between nodes. However, relying on a single message
+stream can result in scalability issues and bottlenecks, as one application would
+need to wait for another to finish utilizing the stream before being able to send
+and receive its own messages.
+
+To overcome this issue, network connections employ stream multiplexing, which allows
+for the creation of multiple virtual connections within a single connection. This
+enables nodes to send multiple streams of messages over separate virtual connections,
+providing a scalable solution that eliminates the bottleneck created by a
+single stream. An example is when HTTP/2 introduced streams into HTTP,
```

Comment:
```suggestion
single stream. 
Two libp2p peers may have a single TCP connection and use different port numbers to distinguish streams.
Then different applications/processes like Kademlia or GossipSub used by an application like IPFS would get their own stream of data and make transmission more efficient.
Stream multiplexing makes it so that applications or protocols running on top of libp2p think that they’re the only ones running on that connection.
Another example is when HTTP/2 introduced streams into HTTP,
```

Reply:
We can add an example, but to the previous point, I'd prefer introducing stream muxing in a way that is generic (not specific to libp2p), mention the case with HTTP/2, and then lead into libp2p and an example with libp2p.

Reply:
Understood, feel free to ignore